snuba/admin/cardinality_analyzer/cardinality_analyzer.py

```diff
@@ -29,13 +31,16 @@ def run_metrics_query(query: str, user: str) -> ClickhouseResult:
     table in ClickHouse. `user` param is necessary for audit_log
     decorator.
     """
-    schema = get_storage(StorageKey("generic_metrics_distributions")).get_schema()
-    assert isinstance(schema, TableSchema)
+    storage_keys = {
+        StorageKey("generic_metrics_distributions"),
+        StorageKey("generic_metrics_sets"),
+        StorageKey("generic_metrics_counters"),
+    }
+    schemas = {get_storage(storage_key).get_schema() for storage_key in storage_keys}
     validate_ro_query(
         sql_query=query,
         allowed_tables={
-            schema.get_table_name(),
-            "generic_metric_distributions_aggregated_dist",
+            cast(TableSchema, schema).get_table_name() for schema in schemas
```

Comment:
Any particular reason why we're casting here?

Reply:
Yes, the bloody type checker complains and fails pre-commit